examples/platform/qpg/app/main.cpp

```diff
@@ -184,7 +184,7 @@ CHIP_ERROR CHIP_Init(void)
         goto exit;
     }
 
-#if CONFIG_CHIP_THREAD_SSED
+#if defined(CHIP_DEVICE_CONFIG_ENABLE_SSED) && CHIP_DEVICE_CONFIG_ENABLE_SSED
```

Comment:
_This_ on the other hand looks wrong.  There is no CHIP_DEVICE_CONFIG_ENABLE_SSED so far (before this PR). What we do have is CHIP_DEVICE_CONFIG_THREAD_SSED on some platforms and CONFIG_CHIP_THREAD_SSED on some platforms, and it's a bit of a mess.

But generally:

1. All `CHIP_*CONFIG*` macros must have a specific config header responsible for ensuring they are defined (and pulling in all the headers that might define them).
2. Consumers of such macros should test them with `#if` _and_ include the responsible header.

The setup here, where `defined` is tested, makes it too easy to fail to include the header that actually defines the config and then build different parts of the code with incompatible configurations...

I know the config header setup is really weird; I am only now starting to finally understand how it works.  But the above is how it should work.  ;)

Reply:
This is now tracked in https://github.com/project-chip/connectedhomeip/issues/29624